### PR TITLE
FIO-8810: fixed an issue where the conditional nested form with several nested levels with required fields cannot be submitted when hidden

### PR DESCRIPTION
**Pull request by TanyaGashtold**
## Link to Jira Ticket

https://formio.atlassian.net/browse/FIO-8810

## Description

**What changed?**

Improved the condition check for nested form so that it could define that the nested form inside conditionally hidden nested components (e.g. another nested form or container) is also considered as hidden.

## Dependencies

https://github.com/formio/core/pull/139

## How has this PR been tested?

Manually + tests

## Checklist:

- [x] I have completed the above PR template
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings
- [x] My changes include tests that prove my fix is effective (or that my feature works as intended)
- [x] New and existing unit/integration tests pass locally with my changes
- [ ] Any dependent changes have corresponding PRs that are listed above

### CODE REVIEW

File: test/submission.js [hunk 1]
```diff
@@ -4999,4 +4999,145 @@ module.exports = function(app, template, hook) {
...
+
+  describe('Conditional Nested Forms Submissions', function () {
+    it('Sets up a default project', function (done) {
+      var owner =
+        app.hasProjects || docker ? template.formio.owner : template.users.admin;
+      helper = new Helper(owner);
+      helper.project().execute(done);
+    });
+
+    it('Create the child form1', (done) => {
```
Comment (johnformio) [inline test/submission.js:5011]:
```suggestion
    before('Create the child form1', (done) => {
```

File: test/submission.js [hunk 2]
```diff
@@ -4999,4 +4999,145 @@ module.exports = function(app, template, hook) {
...
+            key: 'submit',
+            disableOnInvalid: true,
+            input: true,
+            tableView: false,
+          },
+        ])
+        .execute(done);
+    });
+
+    it('Create the child form2', (done) => {
```
Comment (johnformio) [inline test/submission.js:5038]:
```suggestion
    before('Create the child form2', (done) => {
```

File: test/submission.js [hunk 3]
```diff
@@ -4999,4 +4999,145 @@ module.exports = function(app, template, hook) {
         });
     });
   });
+
+  describe('Conditional Nested Forms Submissions', function () {
+    it('Sets up a default project', function (done) {
```
Comment (johnformio) [inline test/submission.js:5004]: All setup steps should be done in a before or beforeAll statement. This allows you to run any test in isolation.

```suggestion
    before('Sets up a default project', function (done) {
```

**Review by johnformio — CHANGES_REQUESTED** [review]
just a few changes to your test.  I'm working towards getting our tests to be more usable in our development workflow and the first part of that is to make sure you can any single test in isolation.

File: test/submission.js (hunk 2 above)
Comment (TanyaGashtold) [reply test/submission.js]: changed

File: test/submission.js (hunk 3 above)
Comment (TanyaGashtold) [reply test/submission.js]: changed

File: test/submission.js (hunk 1 above)
Comment (TanyaGashtold) [reply test/submission.js]: changed

**Review by johnformio — APPROVED** [review]

**Pull request merged**